### PR TITLE
Arb/tokenfactory

### PR DESCRIPTION
**Pull request by 0xArbi**

### CODE REVIEW

File: cmd/starsd/cmd/init.go [hunk 1]
```diff
@@ -150,7 +152,14 @@ func InitCmd(mbm module.BasicManager, defaultNodeHome string) *cobra.Command {
 				return fmt.Errorf("genesis.json file already exists: %v", genFile)
 			}
 
-			appState, err := json.MarshalIndent(mbm.DefaultGenesis(cdc), "", " ")
+			gen := mbm.DefaultGenesis(cdc)
+			sgDefaultGen := sg.NewDefaultGenesisState(cdc)
+			gen["tokenfactory"] = sgDefaultGen["tokenfactory"]
+			// for key := range sgDefaultGen {
```
Comment (shanev) [inline cmd/starsd/cmd/init.go:158]: Is this commented code here for a reason?

**Review by shanev — CHANGES_REQUESTED** [review]
This looks good to me. Are there any custom governance proposals for Token Factory that can be wired up to the gov module?

File: cmd/starsd/cmd/init.go (hunk 1 above)
Comment (0xArbi) [reply cmd/starsd/cmd/init.go:158]: By default the token factory genesis data encodes a denom creation fee of `1stake`, which we don't want. I couldn't for the life of me figure out how to override this genesis value without these ugly changes. 

Happy to pair with someone to figure it out.

**Comment by 0xArbi** [comment]
> This looks good to me. Are there any custom governance proposals for Token Factory that can be wired up to the gov module?

The only change that can be made is to the `DenomCreationFee`, which can be done via a param change. Doesn't need any specific code on our side I believe

**Comment by jhernandezb** [comment]
We will be revising this since there is conflicts with custom bindings

**Pull request closed without merging**